tests/behavior/transforms/remap.toml

```diff
@@ -1894,3 +1894,26 @@
     [[tests.outputs.conditions]]
       type = "remap"
       source = '.a == "A long multiline string"'
+
+[transforms.remap_function_length]
+  inputs = []
+  type = "remap"
+  source = '''
+    .a = length(parse_json(.a))
+    .b = length(parse_json(.b))
+    .c = length(parse_json(.c))
+  '''
+[[tests]]
+  name = "remap_function_length"
+  [tests.input]
+    insert_at = "remap_function_length"
+    type = "log"
+    [tests.input.log_fields]
+    a = '[1, 2, 3]'
+    b = '{"foo":"bar","lorem":{"ipsum": "sicut dolor}}'
+    c = '[[1, 2, 3], [4, 5, 6], []]'
+  [[tests.outputs]]
+    extract_from = "remap_function_length"
+    [[tests.outputs.conditions]]
+      type = "remap"
+      source = '.a == 3 && .b == 2 && .c == 3'
```

Comment:
nit: I mentioned to @FungusHumungus before, as a reader, it might be easier to parse a test written like this:

```rust
[.a, .b, .c] == [3, 2, 3]
```

But, it's just a minor comment, and I'm not even sure I actually prefer that style 😄

Reply:
For a split second I thought you were suggesting that VRL supports Elixir-style pattern matching and I was like 🤯

Reply:
Haha. It actually wouldn't be _that_ difficult to add support with the current set-up, but it's definitely something that requires some understanding to grasp, so let's not do that... yet 😛